pywavefront/visualization.py

```diff
@@ -31,87 +31,109 @@
 # POSSIBILITY OF SUCH DAMAGE.
 # ----------------------------------------------------------------------------
-
+import pyglet
 from pyglet.gl import *
 
-from pywavefront.material import Material
+from pywavefront import Wavefront
 from pywavefront.mesh import Mesh
-from pywavefront.texture import Texture
-
-class _Material(object):
-    """
-    Overrides the drawing routines for the Material class.
-    This class should not be used directly. Its purpose is to shadow
-    the methods to be overriden.
-    """
-
-    def gl_light(self, lighting):
-        """Return a GLfloat with length 4, containing the 4 lighting values."""
-        return (GLfloat * 4)(*(lighting))
-
-    def draw(self, face=GL_FRONT_AND_BACK):
-        if self.texture:
-            self.texture.draw()
-        else:
-            glDisable(GL_TEXTURE_2D)
-
-        glMaterialfv(face, GL_DIFFUSE, self.gl_light(self.diffuse))
-        glMaterialfv(face, GL_AMBIENT, self.gl_light(self.ambient))
-        glMaterialfv(face, GL_SPECULAR, self.gl_light(self.specular))
-        glMaterialfv(face, GL_EMISSION, self.gl_light(self.emissive))
-        glMaterialf(face, GL_SHININESS, self.shininess)
-
-        if self.gl_floats is None:
-            self.gl_floats = (GLfloat * len(self.vertices))(*self.vertices)
-            self.triangle_count = len(self.vertices) / 8
-        glInterleavedArrays(GL_T2F_N3F_V3F, 0, self.gl_floats)
-        glDrawArrays(GL_TRIANGLES, 0, int(self.triangle_count))
-
-
-setattr(Material, "gl_light", _Material.gl_light)
-setattr(Material, "draw", _Material.draw)
-
-
-class _Mesh(object):
-    """
-    Overrides the drawing routines for the Mesh class.
-    This class should not be used directly. Its purpose is to shadow
-    the methods to be overriden.
-    """
-    def draw(self):
-        glPushClientAttrib(GL_CLIENT_VERTEX_ARRAY_BIT)
-        glPushAttrib(GL_CURRENT_BIT | GL_ENABLE_BIT | GL_LIGHTING_BIT)
-        glEnable(GL_CULL_FACE)
-        glCullFace(GL_BACK)
-        for material in self.materials:
-            material.draw()
-        glPopAttrib()
-        glPopClientAttrib()
-
-setattr(Mesh, "draw", _Mesh.draw)
-
-
-class _Texture(object):
-
-    def init(self, path):
-        self.image_name = path
-        self.load_image()
-
-    def draw(self):
-        if not self.image:
-            self.load_image()
-
-        glEnable(self.image.target)
-        glBindTexture(self.image.target, self.image.id)
-        gl.glTexParameterf(self.image.target,
-                           gl.GL_TEXTURE_WRAP_S, gl.GL_CLAMP)
-        gl.glTexParameterf(self.image.target,
-                           gl.GL_TEXTURE_WRAP_T, gl.GL_CLAMP)
-
-    def load_image(self):
-        self.image = pyglet.image.load(self.image_name).texture
-        self.verify_dimensions()
-
-setattr(Texture, "__init__", _Texture.init)
-setattr(Texture, "draw", _Texture.draw)
-setattr(Texture, "load_image", _Texture.load_image)
+
+
+def draw(instance):
+    """Generic draw function"""
```

Comment:
I wonder if there's a way to harmonize these. Example:

```
try:
    draw_meshes(instance)
except:
   raise ValueError(...
```

Reply:
Not entirely sure what you mean 😃

Reply:
There's a lot if `if`/`then`. Frequently that could be more concisely expressed as either a dict (dict key is the type, dict value is the handler), or a try/except.

Reply:
I removed a few unnecessary `if`s. If you still think it needs improvement, just poke

Reply:
👍